addons/web_editor/controllers/main.py

```diff
@@ -301,130 +301,164 @@ def get_image_info(self, image_id=None, xml_id=None):
-                        module = url_info["module"]
-                        module_path = get_module_path(module)
-                        module_resource_path = get_resource_path(module, url_info["resource_path"])
-                        if module_path and module_resource_path:
-                            module_path = os.path.join(os.path.normpath(module_path), '') # join ensures the path ends with '/'
-                            module_resource_path = os.path.normpath(module_resource_path)
-                            if module_resource_path.startswith(module_path):
-                                with open(module_resource_path, "rb") as f:
-                                    content = f.read()
-
-                    bundle_data[1][i] = dict(
-                        url = "/%s/%s" % (url_info["module"], url_info["resource_path"]),
-                        arch = content,
-                        customized = url_info["customized"],
-                    )
+            scss_files_data_by_bundle = self.load_resources('scss', views, bundles_restriction, only_user_custom_files)
 
         return dict(
-            views = get_views and views or [],
-            scss = get_scss and scss_files_data_by_bundle or [],
+            views=get_views and views or [],
+            scss=get_scss and scss_files_data_by_bundle or [],
+            js=get_js and js_files_data_by_bundle or [],
         )
 
-    def save_scss_view_hook(self):
+    def load_resources(self, file_type, views, bundles_restriction, only_user_custom_files):
+        files_data = []
+        resources_type_info = {'t_call_assets_attribute': 't-js', 'mimetype': 'text/javascript'}
+        if file_type == 'scss':
+            resources_type_info = {'t_call_assets_attribute': 't-css', 'mimetype': 'text/scss'}
+
+        # Compile regex outside of the loop
+        # This will used to exclude library scss files from the result
+        excluded_url_matcher = re.compile("^(.+/lib/.+)|(.+import_bootstrap.+\.scss)$")
+
+        # Load already customized files attachments
+        custom_url = self._make_custom_scss_or_js_file_url("%%.%%", "%%")
+        custom_attachments = self.get_custom_attachment(custom_url, op='=like')
+
+        custom_files = [
+            '/website/static/src/scss/user_custom_bootstrap_overridden.scss',
+            '/website/static/src/scss/user_custom_rules.scss',
+            '/website/static/src/js/user_custom_javascript.js'
+        ]
+
+        # First check the t-call-assets used in the related views
+        url_infos = dict()
+        for v in views:
+            for asset_call_node in etree.fromstring(v["arch"]).xpath("//t[@t-call-assets]"):
+                if asset_call_node.get(resources_type_info['t_call_assets_attribute']) == "false":
+                    continue
+                asset_name = asset_call_node.get("t-call-assets")
+
+                # Loop through bundle files to search for file info
+                _files_data = []
```

Comment:
Same comment that the conversation you resolved: this should be `files_data`, same convention as before